File: epi_judge_java/epi/ReverseBits.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class ReverseBits {

    @EpiTest(testDataFile = "reverse_bits.tsv")
    public static long reverseBits(long x) {
        int left = 0;
        int right = 63;
        while (left < right) {
            x = SwapBits.swapBits(x, left++, right--);
        }

        return x;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "ReverseBits.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_python/reverse_bits.py
from swap_bits import swap_bits
from test_framework import generic_test


def reverse_bits(x: int) -> int:
    left = 0
    right = 63
    while left < right:
        x = swap_bits(x, left, right)
        left += 1
        right -= 1
    return x


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('reverse_bits.py', 'reverse_bits.tsv',
                                       reverse_bits))
